src/memex/search/parser.py

```diff
@@ -76,6 +76,31 @@ def parse(q):
     return MultiDict([m for m in parse_results if isinstance(m, Match)])
 
 
+def unparse(q):
+    """
+    Turn a dict-like object into a Lucene-like query string.
+
+    This can be considered the reverse of the
+    :py:func:`memex.search.parser.parse` function, as it can be used to
+    transform the MultiDict returned from that function back into a string
+    query.
+    """
+    terms = []
+
+    try:
+        items = q.iteritems(multi=True)
```

Comment:
That is a *very* good point. Good spot. Given all the tests are passing I guess that `iteritems` from `webob.multidict.MultiDict` does "the right thing". I'll check.